test/fixtures/servers/zookeeper.js

```diff
@@ -7,9 +7,38 @@
 
     var Generic = require('./generic');
     var util = require('util');
+    var exec = require('child_process').exec;
+
+    var service = 'zkServer.sh';
 
     var Zookeeper = function () {
-        Generic.apply(this, ['zkServer.sh', []]);
+        Generic.apply(this, [service, ['start']]);
+        // Zookeeper happily outputs to stderr, so, shut it up
+        this.mute = true;
+
+        var derefStart = this.start;
+
+        this.start = function () {
+            var passargs = arguments;
+            exec('which ' + service, function (error/*, stdout, stderr*/) {
+                // Assume brew osx install in case we didn't find it
+                if (error) {
+                    service = 'zkServer';
+                }
+                // Rebind now
+                this.setCommand(service);
+                // And start
+                derefStart.apply(this, passargs);
+            }.bind(this));
+            return this;
+        };
+
+        this.stop = function () {
+            Generic.apply(this, [service, ['stop']]);
+            this.mute = true;
+            this.start();
```

Comment:
I know - confusing enough...

There is two different process here - `zkserver start` and `zkserver stop` and both need to be "started"...